crates/matrix-sdk-base/src/store/memory_store.rs

```diff
@@ -461,6 +461,27 @@ impl MemoryStore {
             }
         }
 
+        for (room_id, redactions) in &changes.redactions {
+            let mut room_version = None;
+            if let Some(room) = self.room_state.get_mut(room_id) {
+                for mut ref_room_mu in room.iter_mut() {
+                    for mut ref_evt_mu in ref_room_mu.value_mut().iter_mut() {
```

Comment:
I don't think these two `&mut`s need to have the extra `mut` of the binding.

Reply:
Unfortunately, they do:
```
   Compiling matrix-sdk-base v0.5.1 (/home/ben/Dev/matrix/matrix-rust-sdk/crates/matrix-sdk-base)
error[E0596]: cannot borrow `ref_room_mu` as mutable, as it is not declared as mutable
   --> crates/matrix-sdk-base/src/store/memory_store.rs:470:39
    |
469 |                 for ref_room_mu in room.iter_mut() {
    |                     ----------- help: consider changing this to be mutable: `mut ref_room_mu`
470 |                     for ref_evt_mu in ref_room_mu.value_mut().iter_mut() {
    |                                       ^^^^^^^^^^^^^^^^^^^^^^^ cannot borrow as mutable

error[E0596]: cannot borrow `ref_evt_mu` as mutable, as it is not declared as mutable
   --> crates/matrix-sdk-base/src/store/memory_store.rs:471:39
    |
470 |                     for ref_evt_mu in ref_room_mu.value_mut().iter_mut() {
    |                         ---------- help: consider changing this to be mutable: `mut ref_evt_mu`
471 |                         let raw_evt = ref_evt_mu.value_mut();
    |                                       ^^^^^^^^^^^^^^^^^^^^^^ cannot borrow as mutable

   [ .... ]
   Compiling matrix-sdk-base v0.5.1 (/home/ben/Dev/matrix/matrix-rust-sdk/crates/matrix-sdk-base)
error[E0596]: cannot borrow `ref_room_mu` as mutable, as it is not declared as mutable
   --> crates/matrix-sdk-base/src/store/memory_store.rs:470:43
    |
469 |                 for ref_room_mu in room.iter_mut() {
    |                     ----------- help: consider changing this to be mutable: `mut ref_room_mu`
470 |                     for mut ref_evt_mu in ref_room_mu.value_mut().iter_mut() {
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^ cannot borrow as mutable

For more information about this error, try `rustc --explain E0596`.
error: could not compile `matrix-sdk-base` due to previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `matrix-sdk-base` due to previous error
```

Reply:
Ah, `DashMap::iter_mut()` doesn't return `&mut T`, it returns a `RefMutMulti` type which has this `fn value_mut(&mut self)`. Okay then.

Reply:
yeah, it's a bit weird of an API, but I think it is about pushing the write-lock until the last possible moment ... but it leads to this sort of ugliness in the using syntax...